ignition/ignition.go

```diff
@@ -0,0 +1,24 @@
+package ignition
+
+import (
+	"encoding/json"
+	"gopkg.in/yaml.v2"
+
+	"github.com/giantswarm/microerror"
+)
+
+func ConvertTemplatetoJSON(dataIn []byte) ([]byte, error) {
```

Comment:
It is used for tests only for now. But that’s also going to be used in operators for rendering final ignition config as it has to be json

Reply:
Then we should render it as a json and make this private. No need to call `ConvertTemplatetoJSON` in every single operator.

Reply:
I didn’t get it. We need to render templates first with operators parameters and after having complete template in yaml - validate it through ignition schema and marshal to json. How can I do that without calling it in operators?

Reply:
I think this can be done as a single function call in k8scloudconfig:

> render templates first with operators parameters and after having complete template in yaml - validate it through ignition schema and marshal to json.

Reply:
> I think this can be done as a single function call in k8scloudconfig:

Probably true, but I think that brings some mess. With rendering files we are expecting yaml template rendered into yaml config. And ignition validation is a separate stage. The template can be valid from the go perspective and not valid from the ignition spec perspective. So separate validation makes sense to me. imho of course

Reply:
I 100% agree it makes total sense to make those steps separate and have them tested separately.

But I don't think we need to expose that function unless I'm missing something. I'd just make it private.

If this is validation only then I would expose only `Valid(yaml []byte) error`. It can convert it to json under the hood or w/e.

Maybe it doesn't work with additional files like extra units. Let's not talk about this more, maybe when I see the code in one of the operators it's clearer.

Reply:
It is exported because ignition related things are not part of a package. Not sure if that is the proper way. But ignition spec is changed not that often (once in ~1-2 years). So copying that with every cloudconfig package is overhead.